### PR TITLE
Rename `Stake` to `Weight`

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR replaces outdated references to `Stake` in the protocol software with `Weight`. Currently identities and related models have a `Stake` field which actually represents their abstract voting weight rather than their concrete number of staked tokens. We also use "staked" terminology frequently to refer to a state of authorization to participate in the protocol, which does not directly correspond to a node being staked or not. 

* Rename `Stake` struct fields, functions, variables, etc. (which actually currently refer to weight) to `Weight`
  * Add backward-compatible decoding for `Identity`, `NodeInfoPub`, `NodeConfig` (translate `Stake` field to `Weight`)
* Replace "staked/unstaked" terminology, where it refers to "a unejected member of the current epoch with non-zero weight", with "authorized/unauthorized"
* Replace CLI flags incorrectly referencing `stake` (old flags continue to work, but print a warning and are marked as deprecated)
* Left unstaked vs staked access node terminology as-is

## Review Guide
* [First rename pass, focussing on fields, functions, variables](https://github.com/onflow/flow-go/pull/2090/files/ffa2a02496f17f3ac555c3e4ade1984048ea1043) - should not change functionality, only names
* [Second rename pass, focussing on comments, string variables](https://github.com/onflow/flow-go/pull/2090/files/8cbacfa0caf76d4f2c81ab4516d7fafb13388d85..048a108bf24838e583df858a3b8160bf48cfdc3f) - should not change functionality, only names
* [Changes to bootstrap CLI](https://github.com/onflow/flow-go/pull/2090/commits/8cbacfa0caf76d4f2c81ab4516d7fafb13388d85) - includes renamed flags (should be backward-compatible)
* [Changes to integration tests & localnet](https://github.com/onflow/flow-go/pull/2090/files/b0267f990c950c419140df4d68954bed5c546b49..09eb8ffb851c4f3f934613225bd47b36bbc7d4e8) - modifies state bootstrapping logic, functionality should be the same
* [Implementing backward-compatible decoding for structures with Stake field](https://github.com/onflow/flow-go/pull/2090/files/09eb8ffb851c4f3f934613225bd47b36bbc7d4e8..6e03f271a24196d1f6d51b84e1d34aa360813867) - when decoding identity structures, we should translate encoded `Stake` field into `Weight`

### CODE REVIEW

**Comment by AlexHentschel** [comment]
Just saw this:
https://github.com/onflow/flow-go/blob/32cf6152010cc9a56cf815781131e4a553a56ba6/consensus/hotstuff/committees/leader/leader_selection.go#L28-L29

Would you mind updating the goDoc to:
```
IsInvalidViewError returns whether or not the input error is an invalid view error.
```
(typo in `Viw`)

**Review by zhangchiqing — APPROVED** [review]
Nice work  👍

**Review by AlexHentschel — APPROVED** [review]
Thanks Jordan for cleaning this up. Great!

I came accross the few following occurrences of "stake" in the code base:
* https://github.com/onflow/flow-go/blob/315866ba9553808f431ac09f3c92e5c499a4f775/consensus/hotstuff/committees/leader/leader_selection_test.go#L413-L414
* https://github.com/onflow/flow-go/blob/315866ba9553808f431ac09f3c92e5c499a4f775/cmd/bootstrap/utils/unittest.go#L13

File: cmd/bootstrap/cmd/final_list.go [hunk 1]
```diff
@@ -45,92 +45,105 @@ func addFinalListFlags() {
...
-		if matchingNode.Role != stakedNode.Role {
-			log.Error().Str("staked node", stakedNode.NodeID.String()).
-				Str("staked node-type", stakedNode.Role.String()).
-				Str("node", matchingNode.NodeID.String()).
-				Str("node-type", matchingNode.Role.String()).
-				Msg("node type does not match")
+		if matchingNode.Role != registeredNode.Role {
+			log.Error().
+				Str("onchain node id", registeredNode.NodeID.String()).
+				Str("onchain node role", registeredNode.Role.String()).
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/final_list.go:102]:
```suggestion
				Str("registered node id", registeredNode.NodeID.String()).
				Str("registered node role", registeredNode.Role.String()).
```

File: cmd/bootstrap/cmd/final_list.go [hunk 2]
```diff
@@ -45,92 +45,105 @@ func addFinalListFlags() {
...
+				Str("local node role", matchingNode.Role.String()).
+				Msg("node role does not match")
 		}
 
-		// No need to error adderss, and public keys as they may be different
-		// we only keep node-id constant through out sporks
+		if matchingNode.Address != registeredNode.Address {
+			log.Error().
+				Str("onchain node id", registeredNode.NodeID.String()).
+				Str("onchain node address", registeredNode.Address).
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/final_list.go:111]:
```suggestion
				Str("registered node id", registeredNode.NodeID.String()).
				Str("registered node address", registeredNode.Address).
```

**Pull request merged**